### PR TITLE
Add functionality for third-party parameter sets

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
This PR adds: a `cookiecutter.json` configuration to ask users if they wish to register a third-party parameter set, a basic example of a parameter set, and an entry point in `pyproject.toml` for using it with PyBaMM

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
I shall be adding documentation and tests for the same in further pull requests, but I tested this with an example project and it works as intended (the parameter set is defined in `src/pybamm_cookiecutter/my_parameters.py`).

in `pyproject.toml`

```toml
[project.entry-points.pybamm_parameter_sets]
custom_parameters = "src.pybamm_cookiecutter.my_parameters:get_parameter_values"
```
and testing it as follows:

```python
>>> pybamm.parameter_sets.get_docstring("custom_parameters")

'\nExample docstring for a third-party parameter set titled `my_parameters`.\n\nReferences\n----------\n\nAdd references to the relevant papers here in this format, to be parsed by\nsphinxcontrib.bibtex in the documentation through a CITATIONS.bib file. You\nmay refer to the documentation for this extension at\nhttps://sphinxcontrib-bibtex.readthedocs.io/en/latest/\n\n:footcite:`van1995python`\n\n'
```

**Review by Saransh-cpp — APPROVED** [review]
Loos good, thanks, @agriyakhetarpal!

File: {{cookiecutter.project_name}}/src/{{cookiecutter.__project_slug}}/{% if cookiecutter.parameter_set=='True' %}my_parameters.py{% endif %} [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Copyright (c) {{ cookiecutter.__year }} {{ cookiecutter.full_name }}. All rights reserved.
+
+{{ cookiecutter.project_name }}: {{ cookiecutter.project_short_description }}
+"""
+
+import pybamm
+import numpy as np
+
+# Extra dependencies may be imported here.
+
```
Comment (tinosulzer) [inline {{cookiecutter.project_name}}/src/{{cookiecutter.__project_slug}}/{% if cookiecutter.parameter_set=='True' %}my_parameters.py{% endif %}:11]: could you add an example function and scalar parameter?

Comment (agriyakhetarpal) [reply {{cookiecutter.project_name}}/src/{{cookiecutter.__project_slug}}/{% if cookiecutter.parameter_set=='True' %}my_parameters.py{% endif %}:11]: I added a sample function with scalar parameters from the OKane2022 parameter set, let me know if that suffices

**Comment by agriyakhetarpal** [comment]
I will close this to allow @santacodes to work on this further. He will be looking to start with setting up some testing infrastructure and CI/CD, some additional build backends, and he will then proceed to add some parameter sets and a data file plugin (subject to further discussion on the design for the latter).

**Pull request closed without merging**